### PR TITLE
CSCQVAIN-81: Navigation improvements

### PR DESCRIPTION
**Pull request by jppiiroinen**
This will hide the navigation bar when there is the anonymous user. This also includes minor cosmetic changes to typos and words.

### CODE REVIEW

File: src/navigation.vue [hunk 1]
```diff
@@ -65,14 +65,14 @@ Weekdays from 8:30 AM to 4 PM" href="mailto:servicedesk@csc.fi?subject=Fairdata%
...
 			<b-nav-toggle target="app-subbar-collapse"></b-nav-toggle>
 			<b-collapse is-nav id="app-subbar-collapse">
 				<transition name="fade" tag="b-navbar-nav">
 					<b-nav-text v-if="$auth.loading.state" key="loading" class="load-placeholder"></b-nav-text>
-					<b-navbar-nav v-else key="links">
-						<b-nav-item v-if="$route.path !== '/datasets'" key="datasets" to="/datasets">My Datasets</b-nav-item>
-						<b-nav-item v-else key="editor" :to="editorUrl">Editor</b-nav-item>
+					<b-navbar-nav v-else-if="$auth.loggedIn" key="links">
+						<b-nav-item key="datasets" to="/datasets">Datasets</b-nav-item>
+						<b-nav-item key="editor" to="/dataset/new">New Dataset</b-nav-item>
```
Comment (jppiiroinen) [inline src/navigation.vue]: this line?

Comment (jppiiroinen) [reply src/navigation.vue]: I will revert the editorUrl thing

Comment (jppiiroinen) [reply src/navigation.vue]: fixed with another fix.

**Comment by jppiiroinen** [comment]
we iterated this with @Kauhia in private chat

**Review by Kauhia — APPROVED** [review]

**Review by Kauhia — APPROVED** [review]

**Pull request merged**